### PR TITLE
[Enhancement] Lazy load components

### PR DESCRIPTION
**Pull request by jayaike**
Issue: https://github.com/Cloud-CV/EvalAI-ngx/issues/257

I have lazy-loaded all the routing modules. Below are the stats.

# Development

## Optimization ------- 72.7%

# Production

## Optimization ------- 42.8%

# Methodology
### Step 1: Feature Modules
I broke up all the components into feature modules in order to import them lazily. I divided them by purpose and route.

### Step 2: Import Lazily (Change Routes)
I had to change all the routes to lazily import the feature modules. This way the user only loads the modules he/she uses at the time of need.

### Step 3: Change Method of lazy-loading to `PreloadAllModules`
Basically, what `PreloadAllModules` does is it waits for the initial component to load then it starts loading other links so as to reduce the loading time when navigating to other routes.

As shown above, this new way of loading modules in EvalAI-ngx will increase the loading time of EvalAI-ngx and also improve User Experience by making the webpage smoother because it does not have all the modules loaded at once.

Although it took a whole night 🤣😂....It was worth it!

@Ram81 @RishabhJain2018 @pushkalkatara @Sanji515 @lunayach @vkartik97 Please let me know your thoughts.

### CODE REVIEW

**Comment by Sanji515** [comment]
Screenshot from 2020-01-16 02-18-32

@nsjcorps I'm getting this error, can you please take a look at this?

**Comment by jayaike** [comment]
I guess this should be the key https://github.com/angular/angular-cli/issues/14566

**Comment by jayaike** [comment]
It's as a result of the lazy modules. The syntax is correct though. @Sanji515

**Comment by jayaike** [comment]
Do you use yarn or npm. I use yarn
